osu.Game/Scoring/ScoreInfo.cs

```diff
@@ -237,8 +238,8 @@ public bool Equals(ScoreInfo other)
             if (ID != 0 && other.ID != 0)
                 return ID == other.ID;
 
-            if (OnlineScoreID.HasValue && other.OnlineScoreID.HasValue)
-                return OnlineScoreID == other.OnlineScoreID;
+            if (OnlineID > 0)
```

Comment:
The online ID of the other score should matter as well?
```suggestion
            if (OnlineID > 0 && other.OnlineID > 0)
```

Reply:
I'm actually not even sure if we want this here. It doesn't match the new equality methods we decided on for `Beatmap` and `BeatmapSet`. Let me see if I can isolate usages of this and remove it instead.